### PR TITLE
Implement necessary contract method `begin_chain` for `evm-bully`.

### PR DESCRIPTION
**Pull request by frankbraun**
Issues #1 and #2.

### CODE REVIEW

**Comment by frankbraun** [comment]
Most work related to this was done on on the [evm-bully](https://github.com/aurora-is-near/evm-bully) side.

Regarding issue #1 everything is implemented, issue #2 needs more work. With this PR we can now replay Görli up to block 12842 where we fail with:

```
Transaction Id B4cfxkbuPG2xjSUzbxzVY7WDVT2udaSXdyajDZ2DwBjn
{
  "Failure": {
    "ActionError": {
      "index": 0,
      "kind": {
        "FunctionCallError": {
          "ExecutionError": "Exceeded the prepaid gas."
        }
      }
    }
  }
}
```
I would merge this now and do further work with additional PRs.

**Review by joshuajbouw — APPROVED** [review]
Straight forward.

**Review by artob — APPROVED** [review]
LGTM :+1:

File: src/types.rs [hunk 1]
```diff
@@ -10,8 +10,8 @@ use crate::sdk;
 
 pub type AccountId = String;
 pub type RawAddress = [u8; 20];
-pub type RawU256 = [u8; 32];
-pub type RawH256 = [u8; 32];
+pub type RawU256 = [u8; 32]; // Little-endian large integer type.
```
Comment (birchmd) [inline src/types.rs:13]: Is it little endian? I thought EVM was always big endian

Comment (frankbraun) [reply src/types.rs:13]: Yes, it's pulled from here: https://docs.rs/ethereum-types/0.1.4/ethereum_types/

Comment (frankbraun) [reply src/types.rs:13]: Hmm, I'm finding conflicting info on this.

Comment (birchmd) [reply src/types.rs:13]: My question was based on what I read in the Mastering Ethereum book

> All numbers in Ethereum are encoded as big-endian integers

https://github.com/ethereumbook/ethereumbook/blob/develop/06transactions.asciidoc

Comment (joshuajbouw) [reply src/types.rs:13]: In blake2f, they use little endian on parts too, not all of it. *shrug*

Comment (frankbraun) [reply src/types.rs:13]: @birchmd You were right with this. The documentation https://docs.rs/ethereum-types/0.1.4/ethereum_types/ confused me. Web assembly uses little-endian, but the EVM uses big-endian integers. So larger number were encoded wrongly in the `evm-bully` which also caused #122.

**Pull request merged**